src/Protocol.cpp

```diff
@@ -606,3 +600,133 @@ PeriodicUpdate protocol::parseE4Output(uint8_t const* buffer, int bufferSize)
     result.filter_state = state;
     return result;
 }
+
+static Eigen::Matrix3d arrayToCovarianceMatrix(std::array<float, 6> const& v) {
+    Eigen::Matrix3d m = Eigen::Matrix3d::Zero();
+    m(0, 0) = v[0];
+    m(0, 1) = v[1];
+    m(0, 2) = v[2];
+    m(1, 1) = v[3];
+    m(1, 2) = v[4];
+    m(2, 2) = v[5];
+    return m + m.triangularView<Eigen::StrictlyUpper>().transpose().toDenseMatrix();
+}
+
+static Eigen::Matrix4d arrayToCovarianceMatrix(std::array<float, 10> const& v) {
+    Eigen::Matrix4d m = Eigen::Matrix4d::Zero();
+    m(0, 0) = v[0];
+    m(0, 1) = v[1];
+    m(0, 2) = v[2];
+    m(0, 3) = v[3];
+    m(1, 1) = v[4];
+    m(1, 2) = v[5];
+    m(1, 3) = v[6];
+    m(2, 2) = v[7];
+    m(2, 3) = v[8];
+    m(3, 3) = v[9];
+    return m + m.triangularView<Eigen::StrictlyUpper>().transpose().toDenseMatrix();
+}
+
+template<typename ArrayT, size_t Size>
+static Eigen::Matrix<double, Size, 1> arrayToVector(std::array<ArrayT, Size> const& a) {
```

Comment:
```suggestion
static Eigen::Matrix<double, Size, 1> arrayToVectorXd(std::array<ArrayT, Size> const& a) {
```

Reply:
I really don't think that should be a "required change" ...